src/details/ArborX_DetailsVector.hpp

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_DETAILS_VECTOR_HPP
+#define ARBORX_DETAILS_VECTOR_HPP
+
+#include <ArborX_HyperPoint.hpp>
+
+namespace ArborX::Details
+{
+
+// Same as Point, but a different type (can't inherit from Point as it would
+// run into constexpr and inability to aggregate initialize)
+template <int DIM, typename Coordinate = float>
+struct Vector
```

Comment:
If there's a desire to not duplicate the code here, I guess I could live with an alias. I'd like it to be strongly typed, though.

Reply:
In the original implementation we had both the strongly-typed and no duplication (privately inheriting from Point and a couple using-declarations).
What is the constexpr issue you refer to?  How much do we care about aggregate initialization considering it is defined in namespace Details::?

Reply:
If we can get away with deriving and some using declarations, I'd prefer that.
It would also be nice to maybe add some helper functions later, like `Point-Point=Vector` (I see there is makeVector`, though), `Point+Vector=Point`, `Vector+Vector=Vector`.

Reply:
> In the original implementation we had both the strongly-typed and no duplication (privately inheriting from Point and a couple using-declarations). What is the constexpr issue you refer to? How much do we care about aggregate initialization considering it is defined in namespace Details::?

@dalg24 We never used `Vector{1, 2, 3}` with that implementation. When used in testing, it needed `constexpr`. Switching to the general implementation, this seems like a convenient functionality requiring easier test writing. We could indeed use the original implementation and switch to using the constructors (which would need to be marked constexpr).

> It would also be nice to maybe add some helper functions later, like Point-Point=Vector (I see there is makeVector, though), Point+Vector=Point, Vector+Vector=Vector`.

@masterleinad That is not a bad idea.

Reply:
You'd need to write a constructor indeed, at which point I am not sure whether avoiding to write the indexing operator is worth the complexity added.
I suppose I am not asking you to change, just arguing that it can be achieved.